### PR TITLE
Feature/s6 snackbar

### PR DESCRIPTION
**Pull request by federicoboschini**

### CODE REVIEW

File: docs/SNACKBAR.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+![snackbar](snackbar.gif)
+
+To set the activity use the following method:
+
+```java
+snackbar.setActivity(YourActivity.this, NEAR_PERMISSION_REQUEST);
+```
+
+where `NEAR_PERMISSION_REQUEST` is an int value defined by you that will identify the request.
```
Comment (catt-stefano) [inline docs/SNACKBAR.md]: "the request made by the view"

File: docs/SNACKBAR.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+It is **important** that you pass an Activity reference to the snackbar: by doing this, clicking the snackbar button will cause the launch of the permissions request flow.
+
+![snackbar](snackbar.gif)
+
+To set the activity use the following method:
+
+```java
+snackbar.setActivity(YourActivity.this, NEAR_PERMISSION_REQUEST);
```
Comment (catt-stefano) [inline docs/SNACKBAR.md]: I would use a different method name for this, like "bindToActivity" and also add a method to suggest that there's not going to be any memory leak, so an "unbindFromActivity" that a user can call in its activity "onStop"

File: docs/SNACKBAR.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+```
+
+where `NEAR_PERMISSION_REQUEST` is an int value defined by you that will identify the request.
+Then in your activity you should catch the `result` by referring to the same request code.
+
```
Comment (catt-stefano) [inline docs/SNACKBAR.md]: Add a sample of the callback method: should I catch the onActivityResult or the onRequestPermissionsResult? It's not clear from the doc.

File: nearit-ui/src/main/java/com/nearit/ui_bindings/utils/PreRequirementsUtil.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import android.location.LocationManager;
+import android.support.v4.content.ContextCompat;
+
+import it.near.sdk.logging.NearLog;
+
+/**
+ * Created by Federico Boschini on 28/09/17.
+ */
+
+public class PreRequirementsUtil {
```
Comment (catt-stefano) [inline nearit-ui/src/main/java/com/nearit/ui_bindings/utils/PreRequirementsUtil.java:15]: Add a javadoc comment to the public static methods of this class

File: nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="@dimen/nearit_ui_permission_snackbar_button_height"
+    android:paddingLeft="@dimen/nearit_ui_permission_snackbar_button_left_right_padding"
+    android:paddingRight="@dimen/nearit_ui_permission_snackbar_button_left_right_padding"
+    android:paddingTop="@dimen/nearit_ui_permission_snackbar_button_top_bottom_padding"
+    android:paddingBottom="@dimen/nearit_ui_permission_snackbar_button_top_bottom_padding"
+    android:gravity="center"
+    android:background="@drawable/nearit_ui_permission_snackbar_button">
```
Comment (catt-stefano) [inline nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml]: No selector?

File: nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (catt-stefano) [inline nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml]: The button layout is a bit different from the mock, it doesn't seem large enough and the test should be bold

**Review by catt-stefano — CHANGES_REQUESTED** [review]
Aside from the comments on the code, let's remove all instances of "Snackbar" since it's not a snackbar anymore. "NearITPermissionView"

File: docs/SNACKBAR.md (hunk 2 above)
Comment (federicoboschini) [reply docs/SNACKBAR.md]: done. unbindFromActivity() sets the activity to null, can it be the right way to do it?

File: docs/SNACKBAR.md (hunk 1 above)
Comment (federicoboschini) [reply docs/SNACKBAR.md]: done

File: nearit-ui/src/main/java/com/nearit/ui_bindings/utils/PreRequirementsUtil.java (hunk 4 above)
Comment (federicoboschini) [reply nearit-ui/src/main/java/com/nearit/ui_bindings/utils/PreRequirementsUtil.java:15]: done

File: docs/SNACKBAR.md (hunk 3 above)
Comment (federicoboschini) [reply docs/SNACKBAR.md]: done and updated the sample

File: nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml (hunk 5 above)
Comment (federicoboschini) [reply nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml]: done, added the selector

File: nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml (hunk 6 above)
Comment (federicoboschini) [reply nearit-ui/src/main/res/layout/nearit_ui_layout_permission_snackbar_button.xml]: done

**Pull request merged**